src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,202 @@
+import colors from 'src/libs/colors'
+import { useCancellation, useOnMount } from 'src/libs/react-utils'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+export const WdsTroubleshooter = ({ onDismiss, workspaceId, mrgId }) => {
+  const [leoOk, setLeoOk] = useState(null)
+  const [wdsResponsive, setWdsResponsive] = useState(null)
+  const [version, setVersion] = useState(null)
+  const [wdsStatus, setWdsStatus] = useState(null)
+  const [wdsDbStatus, setWdsDbStatus] = useState(null)
+  const [wdsPingStatus, setWdsPingStatus] = useState(null)
+  const [appFound, setAppFound] = useState(null)
+  const [appRunning, setAppRunning] = useState(null)
+  const [proxyUrl, setProxyUrl] = useState(null)
+  const [defaultInstanceExists, setDefaultInstanceExists] = useState(null)
+
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    Ajax(signal).Apps.getV2AppInfo(workspaceId).then(res => {
+      setLeoOk(res)
+      const foundApp = resolveWdsApp(res)
+      setAppFound(foundApp?.appName)
+      setAppRunning(foundApp?.status)
+      setProxyUrl(foundApp?.proxyUrls?.wds)
+      Ajax(signal).WorkspaceData.getVersion(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsResponsive(true)
+        setVersion(res.git?.commit?.id)
+      }).catch(_ => {
+        setWdsResponsive(false)
+        setVersion('unknown')
+      })
+      Ajax(signal).WorkspaceData.getStatus(foundApp?.proxyUrls?.wds).then(res => {
+        setWdsStatus(res.status)
+        setWdsDbStatus(res.components?.db?.status)
+        setWdsPingStatus(res.components?.ping?.status)
+      }).catch(_ => {
+        setWdsStatus('unresponsive')
+        setWdsDbStatus('unknown')
+        setWdsPingStatus('unknown')
+      })
+      Ajax(signal).WorkspaceData.listInstances(foundApp?.proxyUrls?.wds).then(res => {
+        setDefaultInstanceExists(res.includes(workspaceId))
+      }).catch(_ => {
+        setDefaultInstanceExists('unknown')
+      })
+    }).catch(_ => {
+      setLeoOk([])
+      setAppFound('unknown')
+      setAppRunning('unknown')
+      setProxyUrl('unknown')
+      setWdsResponsive('unknown')
+      setVersion('unknown')
+      setWdsStatus('unresponsive')
+      setWdsDbStatus('unknown')
+      setWdsPingStatus('unknown')
+    })
+  })
+
+  const checkIcon = (status, size = 24) => Utils.switchCase(status,
+    ['success', () => icon('success-standard', { size, style: { color: colors.success() }, 'aria-label': 'Validation Success' })],
+    ['failure', () => icon('error-standard', { size, style: { color: colors.danger(0.85) }, 'aria-label': 'Validation Failure' })],
+    ['running', () => icon('loadingSpinner', { size, style: { color: colors.primary() }, 'aria-label': 'Validation Running' })]
+  )
+
+  const clippy = (label, value) => !!value && h(Fragment, [value, h(ClipboardButton, {
+    'aria-label': `Copy ${label} to clipboard`,
+    style: { marginLeft: '1rem' },
+    text: value
+  })])
+
+  const troubleShooterText = {
```

Comment:
I wasn't real sure of a good way to copy all (and just) the text from the whole troubleshooting table, so I've created this handy dictionary for the clipboard icon.  Could potentially refactor to use this or something like it when creating the table to avoid duplication, but am also very interested in other ideas.